python/ray/data/_internal/plan.py

```diff
@@ -618,36 +586,19 @@ def collect_stats(cur_stats):
 
             # Set the snapshot to the output of the final operator.
             self._snapshot_blocks = blocks
-            if not isinstance(blocks, LazyBlockList):
-                self._snapshot_bundle = RefBundle(
-                    tuple(blocks.iter_blocks_with_metadata()),
-                    owns_blocks=blocks._owned_by_consumer,
-                )
+            self._snapshot_bundle = RefBundle(
+                tuple(blocks.iter_blocks_with_metadata()),
+                owns_blocks=blocks._owned_by_consumer,
+            )
             self._snapshot_operator = self._logical_plan.dag
             self._snapshot_stats = stats
             self._snapshot_stats.dataset_uuid = self._dataset_uuid
+        return self._snapshot_bundle
 
-            # In the case of a read-only dataset, we replace the
-            # input LazyBlockList with a copy that includes the
-            # calculated metadata from initializing the InputDataBuffer.
-            if self.is_read_only():
-                self._in_blocks = blocks
-        if isinstance(self._snapshot_blocks, LazyBlockList) and force_read:
-            executed_blocks = self._snapshot_blocks.compute_to_blocklist()
-            # After executing the snapshot blocks, get its updated stats.
-            # The snapshot blocks after execution will contain the execution stats.
-            self._snapshot_stats = self._snapshot_blocks.stats()
-            self._snapshot_blocks = executed_blocks
-            assert not isinstance(executed_blocks, LazyBlockList), type(executed_blocks)
-            self._snapshot_bundle = RefBundle(
-                tuple(executed_blocks.iter_blocks_with_metadata()),
-                owns_blocks=executed_blocks._owned_by_consumer,
-            )
-            self._snapshot_operator = self._logical_plan.dag
-            # When force-read is enabled, we similarly update self._in_blocks.
-            if self.is_read_only():
-                self._in_blocks = self._snapshot_blocks
-        return self._snapshot_blocks
+    @property
+    def has_executed(self) -> bool:
+        """Return whether this plan has been partially or fully executed."""
```

Comment:
since the method returns a bool, the original wording was a bit confusing for me (thought it should return True if fully executed, False if partially executed). just a suggestion:
```suggestion
        """Return True if this plan has been partially or fully executed."""
```
OR
```suggestion
        """Return whether this plan has started execution."""
```